src/writer.rs

```diff
@@ -507,11 +515,25 @@ impl Writer {
         Ok(())
     }
 
+    fn format_declaration<'d, W: ?Sized>(&self, writer: &mut W) -> io::Result<()>
+        where W: Write
```

Comment:
I really appreciate you mirroring the (quite old) style of the existing code!

Someday I'll actually update for modern style and upgrade to Rust 2018, etc.... someday.